cap-containers/map.h

```diff
@@ -165,6 +174,25 @@ static cap_map *cap_map_init(size_t key_size,
 	map->_key_size = key_size;
 	return map;
 }
+static cap_map_iterator *cap_map_iterator_init(cap_map *map){
+	assert(map != NULL);
+	cap_map_iterator *iterator = 
+	    (cap_map_iterator *)CAP_ALLOCATOR(cap_map_iterator, 1);
+	if (!iterator) {
+		fprintf(stderr, "memory allocation failue\n");
+		return NULL;
+	}
+	if (!map){
+		iterator->_current_index = 0;
+		map->_key[0]=NULL;
+		map->_value=NULL;
+	}
+	iterator->_current_index = map->_forward[0];
```

Comment:
This is incorrect. `iterator->_current_index` is of type `size_t`, but you're setting it to of type `cap_map`.

This will not compile.